Public/Src/Engine/Cache/Artifacts/LocalDiskContentStore.cs

```diff
@@ -926,50 +925,58 @@ private void TryFlushPageCacheToFileSystem(string path)
+                                            strict: false);
+                                    identityAndContentInfo = new VersionedFileIdentityAndContentInfo(identity, new FileContentInfo(hash, length));
+                                }
+
+                                if (!trackPath)
+                                {
+                                    // We are not interested in tracking the file (perhaps because it has been tracked, see our handling of copy-file pip),
+                                    // but we need the file length, and potentially need to ensure that the file name casing match.
+                                    return new TrackedFileContentInfo(new FileContentInfo(hash, length), FileChangeTrackingSubscription.Invalid, fileName);
+                                }
+
+                                Contract.Assert(recordPathInFileContentTable && identityAndContentInfo.HasValue);
 
-                        if (!trackPath)
+                                // Note that the identity kind may be Anonymous if we couldn't establish an identity for the target file;
+                                // in that case we still need to call TrackChangesToFile to ensure the tracker latches to disabled.
+                                return TrackChangesToFile(handle, path, identityAndContentInfo.Value, knownFileName: fileName);
+                            });
+
+                        if (attempt.Succeeded)
                         {
-                            // We are not interested in tracking the file (perhaps because it has been tracked, see our handling of copy-file pip),
-                            // but we need the file length, and potentially need to ensure that the file name casing match.
-                            return new TrackedFileContentInfo(new FileContentInfo(hash, length), FileChangeTrackingSubscription.Invalid, fileName);
+                            return new Possible<TrackedFileContentInfo, Failure>(attempt.Result);
                         }
 
-                        Contract.Assert(recordPathInFileContentTable && identityAndContentInfo.HasValue);
-
-                        // Note that the identity kind may be Anonymous if we couldn't establish an identity for the target file;
-                        // in that case we still need to call TrackChangesToFile to ensure the tracker latches to disabled.
-                        return TrackChangesToFile(handle, path, identityAndContentInfo.Value, knownFileName: fileName);
-                    });
-            }
-            catch (BuildXLException ex)
-            {
-                return new RecoverableExceptionFailure(ex);
-            }
+                        Tracing.Logger.Log.RetryTryOpenAndTrackPathAsync(m_loggingContext, path.ExpandedPath);
+                        return new Possible<TrackedFileContentInfo, Failure>(new Failure<string>($"TryOpenAndTrackPathAsync() failed to establish identity and track file: {path.ExpandedPath}"));

```

Comment:
nit: instead of `new Possible(new Failure())` you can just return `new Failure<string>("...")`